pkg/provisioner/ironic/raid.go

```diff
@@ -99,11 +99,26 @@ func buildTargetHardwareRAIDCfg(volumes []metal3v1alpha1.HardwareRAIDVolume) (lo
 			}
 			nameCheckFlags[volume.Name] = index
 		}
+
+		// Check numberOfPhysicalDisks is same as len(physicalDisks)
+		if volume.NumberOfPhysicalDisks != nil && volume.PhysicalDisks != nil {
```

Comment:
I assume you've verified that PhysicalDisks is nil (and not just an empty slice) when it is not specified?

Reply:
Oh, you're right. It's not a pointer, so it will be an empty slice. Let me fix that.

Reply:
Slices can be nil (even if they're not pointers), so it's never clear to me what the deserialization of any particular field will give us without trying it. Checking for zero length is a safe way to cover everything though.